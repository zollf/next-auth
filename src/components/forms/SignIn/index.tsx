import React, { useState } from 'react';
import Router from 'next/router';
import { signIn } from 'next-auth/client';

const SignIn = () => {
  const [name, setName] = useState('');
  const [password, setPassword] = useState('');
  const onSubmit = () => {
    signIn('credentials', { name, password, redirect: false }).then((res) => {
      if (res.error) {
        Router.push(`/signIn?error=${res.error}`);
      } else {
        Router.push('/');
      }
    });
  };

  return (
    <div>
      <section>
        <label htmlFor="username">Username</label>
        <input type="text" name="username" value={name} onChange={(e) => setName(e.target.value)} />
      </section>
      <section>
        <label htmlFor="password">Password</label>
        <input type="password" name="password" value={password} onChange={(e) => setPassword(e.target.value)} />
      </section>
      <button type="button" onClick={onSubmit}>
        Submit
      </button>
    </div>
  );
};

export default SignIn;
