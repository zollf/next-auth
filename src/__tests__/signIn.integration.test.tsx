import React from 'react';
import { ReactWrapper, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import SignIn from '@/pages/signIn';

const component = <SignIn />;

jest.mock('next/router', () => ({
  useRouter: () => ({ query: { error: undefined } }),
}));

describe('<SignIn />', () => {
  let wrapper!: ReactWrapper;
  beforeEach(() => {
    wrapper = mount(component);
  });

  it('matches its snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
