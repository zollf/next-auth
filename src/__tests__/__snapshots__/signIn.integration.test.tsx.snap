// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> matches its snapshot 1`] = `
<SignIn>
  <Wrapper>
    <div
      className="wrapper"
    >
      <h1>
        Sign In
      </h1>
      <SignIn>
        <fieldset>
          <section>
            <label
              htmlFor="username"
            >
              Username
            </label>
            <Input
              name="name"
              onChange={[Function]}
              type="text"
              value=""
            >
              <input
                className="input"
                name="name"
                onChange={[Function]}
                type="text"
                value=""
              />
            </Input>
          </section>
          <section>
            <label
              htmlFor="password"
            >
              Password
            </label>
            <Input
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            >
              <input
                className="input"
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
            </Input>
          </section>
          <Button
            onClick={[Function]}
          >
            <button
              className="button"
              onClick={[Function]}
              type="button"
            >
              Submit
            </button>
          </Button>
        </fieldset>
      </SignIn>
    </div>
  </Wrapper>
</SignIn>
`;
